core/app/assets/javascripts/spree.js.erb

```diff
@@ -51,13 +51,45 @@ Spree.ajax = function(url, options) {
   return $.ajax(url, options);
 };
 
-Spree.routes = {
-  states_search: Spree.pathFor('api/states'),
-  apply_coupon_code: function(order_id) {
-    return Spree.pathFor("api/orders/" + order_id + "/coupon_codes");
+/**
+ * @deprecated Spree.routes will be removed in a future release. Please use Spree.pathFor instead.
+ * See: https://github.com/solidusio/solidus/issues/3405
+ */
+Spree.routesDeprecationProxy = {
+  get: function(obj, prop) {
+    <% if Rails.env != "production" %>
+      console.log("Spree.routes is deprecated, please use pathFor instead. See: https://github.com/solidusio/solidus/issues/3405");
+    <% end %>
+
+    return obj[prop]
+  },
+
+  set: function(obj, prop, value) {
+    <% if Rails.env != "production" %>
+      console.log("Spree.routes is deprecated, please use pathFor instead. See: https://github.com/solidusio/solidus/issues/3405");
+    <% end %>
+
+    obj[prop] = value
+    return value;
   }
 };
 
+if(typeof Proxy == "function") {
+  Spree.routes = new Proxy({
+    states_search: Spree.pathFor('api/states'),
+    apply_coupon_code: function(order_id) {
+      return Spree.pathFor("api/orders/" + order_id + "/coupon_codes");
+    }
```

Comment:
Can we extract this into a separate variable so there's less duplication in the code?

Reply:
Absolutely! Check out the latest version.